src/client_side_reply.cc

```diff
@@ -1486,15 +1487,14 @@ clientReplyContext::identifyStoreObject()
 
     // client sent CC:no-cache or some other condition has been
     // encountered which prevents delivering a public/cached object.
-    // XXX: The above text does not match the condition below. It might describe
-    // the opposite condition, but the condition itself should be adjusted
-    // (e.g., to honor flags.noCache in cache manager requests).
-    if (!r->flags.noCache || r->flags.internal) {
+    if (r->flags.noCache) {
+        // no-cache requests skip Store lookups
+        identifyFoundObject(nullptr, "no-cache");
```

Comment:
flags.noCache is not synonymous with "Cache-Control: no-cache" as this comment might imply. Here, we do not need to (re)document what that flag means:

 ```suggestion
         identifyFoundObject(nullptr, "no-cache"); // skip Store lookup
```

Reply:
Done.